include/mxnet/gpu_swap_history.h

```diff
@@ -39,31 +41,43 @@ class MemHistory {
 
   //std::vector<std::vector<MemRecord> > history 
   //    = std::vector<std::vector<MemRecord> >(NUMBER_OF_GPU);
-  std::vector<std::map<handle_id_t, std::vector<MemRecord> > > history
-      = std::vector<std::map<handle_id_t, std::vector<MemRecord> > >
-      (NUMBER_OF_GPU);
-  std::vector<std::vector<MemRecord> > ordered_history =
-      std::vector<std::vector<MemRecord> >(NUMBER_OF_GPU);
-  std::vector<size_t> record_idx = std::vector<size_t>(NUMBER_OF_GPU);
+  std::vector<std::map<handle_id_t, std::vector<MemRecord> > > history;
+      //= std::vector<std::map<handle_id_t, std::vector<MemRecord> > >
+      //(NUMBER_OF_GPU);
+  std::vector<std::vector<MemRecord> > ordered_history;
+      //=std::vector<std::vector<MemRecord> >(NUMBER_OF_GPU);
+  std::vector<std::list<handle_id_t> > lru_list;
+      //= std::vector<std::list<handle_id_t> >(NUMBER_OF_GPU);
+  std::vector<std::unordered_map<handle_id_t, std::list<handle_id_t>::iterator> >
+      lru_map;
+      //= std::vector<std::unordered_map<handle_id_t,
+      //std::list<handle_id_t>::iterator> >(NUMBER_OF_GPU);
+  std::vector<size_t> record_idx; //= std::vector<size_t>(NUMBER_OF_GPU);
 
   ~MemHistory();
   static MemHistory* Get();
   static std::shared_ptr<MemHistory> _GetSharedRef();
   bool IterationStarted() {return iteration_started_;}
+  bool IsPreRecording() {return pre_recording_;}
   bool IsRecording() {return is_recording_;}
+  void PreRecord(handle_id_t handle_id, record_t operation_id, int device);
   void PutRecord(handle_id_t handle_id, int device, record_t type, size_t size);
+  handle_id_t LRU(std::unordered_set<handle_id_t> handles, int device);
+  handle_id_t NaiveHistoryBased(std::unordered_set<handle_id_t> handles,
```

Comment:
Move these two function declarations to line 72 and add a comment: Swap algorithms declaration.